x-pack/plugins/fleet/server/services/epm/elasticsearch/transform/install.ts

```diff
@@ -117,6 +134,218 @@ export const installTransform = async (
   return { installedTransforms, esReferences };
 };
 
+export const installTransforms = async (
+  installablePackage: InstallablePackage,
+  paths: string[],
+  esClient: ElasticsearchClient,
+  savedObjectsClient: SavedObjectsClientContract,
+  logger: Logger,
+  esReferences?: EsAssetReference[]
+) => {
+  const transformPaths = paths.filter((path) => isTransform(path));
+  // If package contains legacy transform specifications (i.e. with json instead of yml)
+  if (transformPaths.some((p) => p.endsWith('.json')) || transformPaths.length === 0) {
+    return await installLegacyTransforms(
+      installablePackage,
+      paths,
+      esClient,
+      savedObjectsClient,
+      logger,
+      esReferences
+    );
+  }
+
+  // Assuming the package will only contain .yml specifications
+  const installation = await getInstallation({
+    savedObjectsClient,
+    pkgName: installablePackage.name,
+  });
```

Comment:
nit: both `installTransforms` and `installLegacyTransforms` fetch this object. Could we avoid that and only fetch once and pass to both functions?

Reply:
Updated here [`fa7e8e2` (#140046)](https://github.com/elastic/kibana/pull/140046/commits/fa7e8e2dd225c513f5e91aa3f4ec9d917defa57e)